src/sourmash/sig/__main__.py

```diff
@@ -1384,6 +1396,99 @@ def check(args):
         sys.exit(-1)
 
 
+def collect(args):
+    "Collect signature metadata across many locations, save to manifest"
+    # TODO:
+    # test what happens with directories :)
+    set_quiet(False, args.debug)
+
+    if os.path.exists(args.output):
+        if args.merge_previous:
+            pass
+        else:
+            error(f"ERROR: '{args.output}' already exists!")
+            error(f"ERROR: please remove it, or use --merge-previous to merge")
+            sys.exit(-1)
+    elif args.merge_previous:
+        notify(f"WARNING: --merge-previous specified, but output file '{args.output}' does not already exist?")
+
+    # load previous manifest for --merge-previous. This gets tricky with
+    # mismatched manifest types, which we forbid.
+    try:
+        if args.manifest_format == 'sql':
+            # create on-disk manifest
+            from sourmash.index.sqlite_index import SqliteCollectionManifest
+
+            if args.merge_previous:
+                collected_mf = SqliteCollectionManifest.create_or_open(args.output)
+            else:
+                collected_mf = SqliteCollectionManifest.create(args.output)
+        else:
+            # create in-memory manifest that will be saved as CSV
+            assert args.manifest_format == 'csv'
+
+            if args.merge_previous and os.path.exists(args.output):
+                collected_mf = CollectionManifest.load_from_filename(args.output)
+            else:
+                collected_mf = CollectionManifest()
+
+            if not isinstance(collected_mf, CollectionManifest):
+                raise Exception
+    except:
+        error(f"ERROR loading '{args.output}' with --merge-previous. Is it of type {args.manifest_format}?")
+        sys.exit(-1)
+
+    if args.merge_previous:
+        notify(f"merging new locations with {len(collected_mf)} previous rows.")
+
+    # require manifests? yes by default, since generating can be slow.
+    require_manifest = True
+    if args.no_require_manifest:
+        require_manifest = False
+        debug("sig check: manifest will not be required")
+    else:
+        debug("sig check: manifest required")
+
+    n_files = 0
+
+    # load from_file
+    _extend_signatures_with_from_file(args, target_attr='locations')
+
+    # convert to abspath
+    if args.abspath:
+        args.locations = [ os.path.abspath(iloc) for iloc in args.locations ]
+
+    # iterate through, loading all the manifests from all the locations.
+    for n_files, loc in enumerate(args.locations):
+        notify(f"Loading signature information from {loc}.")
+
+        if n_files and n_files % 100 == 0:
+            notify(f'... loaded {len(collected_mf)} sigs from {n_files} files')
```

Comment:
```suggestion
        if n_files % 100 == 0:
            notify(f'... loaded {len(collected_mf)} sigs from {n_files} files')
```
would this allow it to be covered by tests (report on 0th file)?